chain/chain/src/store.rs

```diff
@@ -2205,6 +2250,87 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    // Clearing block data of `block_hash`, for the purpose of undo-block tool
+    pub fn clear_block_data_undo_block(&mut self, block_hash: CryptoHash) -> Result<(), Error> {
+        let mut store_update = self.store().store_update();
+        let block =
+            self.get_block(&block_hash).expect("block data is not expected to be already cleaned");
+
+        // 1. delete TrieChanges
+        for shard_id in 0..block.header().chunk_mask().len() as ShardId {
+            let block_shard_id = get_block_shard_id(&block_hash, shard_id);
+            self.gc_col(DBCol::TrieChanges, &block_shard_id);
+        }
+
+        let height = block.header().height();
+
+        // 2. Delete shard_id-indexed data (Receipts, ChunkExtra, State Headers and Parts)
+        for shard_id in 0..block.header().chunk_mask().len() as ShardId {
+            let block_shard_id = get_block_shard_id(&block_hash, shard_id);
+            self.gc_outgoing_receipts(&block_hash, shard_id);
+            self.gc_col(DBCol::IncomingReceipts, &block_shard_id);
+
+            // delete DBCol::ChunkExtra based on shard_uid since it's indexed by shard_uid in the storage
+            self.gc_col(DBCol::ChunkExtra, &block_shard_id);
+
+            // For incoming State Parts it's done in chain.clear_downloaded_parts()
+            // The following code is mostly for outgoing State Parts.
+            // However, if node crashes while State Syncing, it may never clear
+            // downloaded State parts in `clear_downloaded_parts`.
+            // We need to make sure all State Parts are removed.
+            if let Ok(shard_state_header) = self.chain_store.get_state_header(shard_id, block_hash)
+            {
+                let state_num_parts =
+                    get_num_state_parts(shard_state_header.state_root_node().memory_usage);
+                self.gc_col_state_parts(block_hash, shard_id, state_num_parts)?;
+                let state_header_key = StateHeaderKey(shard_id, block_hash).try_to_vec()?;
+                self.gc_col(DBCol::StateHeaders, &state_header_key);
+            }
+            // delete flat storage columns: FlatStateChanges and FlatStateDeltaMetadata
```

Comment:
@pugachAG @Longarithm flat state column changes are here, please help review

Reply:
Yeah, we don't need anything else. `home_dir` is used to read genesis, `store` - at least for reading state data, `config`... for tweaking execution parameters, caching, etc.